reconcile/utils/vault.py

```diff
@@ -166,7 +195,8 @@ def _read_all_v2(self, path, version):
             raise SecretNotFound(path)
 
         data = secret["data"]["data"]
-        return data
+        secret_verion = secret["data"]["metadata"]["version"]
+        return data, secret_verion
```

Comment:
hmmm, you should not modify the `read_all` but use the new method...

on top of that... logic should not be duplicated. The old method should call the new one returning just the data.